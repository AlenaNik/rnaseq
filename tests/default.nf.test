nextflow_pipeline {

    name "Test pipeline with default settings"
    script "../main.nf"

    test("Params: default") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            def path_not_snapshottable = UTILS.getAllFilesFromDir(params.outdir, null, UTILS.exclusionRegexesForUnstableFileNames)
            def path_snapshottable     = UTILS.getAllFilesFromDir(params.outdir, UTILS.snapshottablePatterns, UTILS.exclusionRegexesForUnstableFileContents)

            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.trace.tasks().size(),
                    UTILS.removeNextflowVersion("$outputDir/pipeline_info/nf_core_rnaseq_software_mqc_versions.yml"),
                    path_not_snapshottable.collect{ file -> path_snapshottable.contains(file) ? null : file.getName()}.findAll{ it },
                    path_snapshottable
                ).match() }
            )
        }
    }

    test("Params: default - stub") {

        options "-stub"

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {

            def path_not_snapshottable = UTILS.getAllFilesFromDir(params.outdir, null, UTILS.exclusionRegexesForUnstableFileNames)
            def path_snapshottable     = UTILS.getAllFilesFromDir(params.outdir, UTILS.snapshottablePatterns, UTILS.exclusionRegexesForUnstableFileContents)

            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.trace.tasks().size(),
                    UTILS.removeNextflowVersion("$outputDir/pipeline_info/nf_core_rnaseq_software_mqc_versions.yml"),
                    path_not_snapshottable.collect{ file -> path_snapshottable.contains(file) ? null : file.getName()}.findAll{ it },
                    path_snapshottable
                ).match() }
            )
        }
    }
}
