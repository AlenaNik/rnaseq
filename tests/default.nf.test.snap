{
    "references": {
        "content": [
            "genome_gfp.fasta:md5,e23e302af63736a199985a169fdac055",
            "genome_gfp.gtf:md5,c98b12c302f15731bfc36bcf297cfe28"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-21T19:22:22.006847"
    },
    "fastqc/trim": {
        "content": [
            "RAP1_IAA_30M_REP1_trimmed_1_val_1_fastqc.html:md5,be3df48707a8e573984eefc5c14a03d5",
            "RAP1_IAA_30M_REP1_trimmed_1_val_1_fastqc.zip:md5,83d97092d77748d5041cbb5d4e138543",
            "RAP1_IAA_30M_REP1_trimmed_2_val_2_fastqc.html:md5,16162c5f476b57fb7b3ac10f3e016b0e",
            "RAP1_IAA_30M_REP1_trimmed_2_val_2_fastqc.zip:md5,fff332745d9186d7ea2c01d9b9aa4442",
            "RAP1_UNINDUCED_REP1_trimmed_trimmed_fastqc.html:md5,bce21c8b85bcc591723927356d17109c",
            "RAP1_UNINDUCED_REP1_trimmed_trimmed_fastqc.zip:md5,a005818480609e511fdc516bde8000c9",
            "RAP1_UNINDUCED_REP2_trimmed_trimmed_fastqc.html:md5,87ec232cb87031475c24820324421dee",
            "RAP1_UNINDUCED_REP2_trimmed_trimmed_fastqc.zip:md5,c714ace4a101fa73462f149ef222003e",
            "WT_REP1_trimmed_1_val_1_fastqc.html:md5,747c6ed7dbb2ddea2a231b9fd8726c4c",
            "WT_REP1_trimmed_1_val_1_fastqc.zip:md5,88e604365901240bb21dd6e8cd803926",
            "WT_REP1_trimmed_2_val_2_fastqc.html:md5,3a0f2595881efe2b1b014f731bb03869",
            "WT_REP1_trimmed_2_val_2_fastqc.zip:md5,dbef90ece043fd51bdaa52658e9ee12c",
            "WT_REP2_trimmed_1_val_1_fastqc.html:md5,bee0baae4f120d669067d2e6824567f6",
            "WT_REP2_trimmed_1_val_1_fastqc.zip:md5,bf4a8372297789a3e4f7662368ac74df",
            "WT_REP2_trimmed_2_val_2_fastqc.html:md5,4cc6923a6f76d7ad99d6727f66bcd094",
            "WT_REP2_trimmed_2_val_2_fastqc.zip:md5,8d0ea8b64a9524a5dfd9ac20de431868"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-21T19:46:35.901089"
    },
    "bbsplit": {
        "content": [
            "RAP1_IAA_30M_REP1.stats.txt:md5,a87292c6f7166096166a9015c2e63dac",
            "RAP1_UNINDUCED_REP1.stats.txt:md5,781e49024150a8313779c574b58ad035",
            "RAP1_UNINDUCED_REP2.stats.txt:md5,e455d0163bf2d988224b0c0092605213",
            "WT_REP1.stats.txt:md5,6eab6936804c1157d5e1c61c0562d7de",
            "WT_REP2.stats.txt:md5,002ddf6b828d34979652a40f05462843"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-21T19:46:35.032801"
    },
    "fastqc/raw": {
        "content": [
            "RAP1_IAA_30M_REP1_raw_1_fastqc.html:md5,b55cd1ea7b117a32c460e1fe6b075929",
            "RAP1_IAA_30M_REP1_raw_1_fastqc.zip:md5,295b8bbb4a9b74c4152d578c959d42d6",
            "RAP1_IAA_30M_REP1_raw_2_fastqc.html:md5,08f90dff9def052b80348669f501c0cc",
            "RAP1_IAA_30M_REP1_raw_2_fastqc.zip:md5,c7ee8233a262dac8a9badea6dc07354a",
            "RAP1_UNINDUCED_REP1_raw_fastqc.html:md5,84540e07fa0432f7f74aaa6ceacfa6d7",
            "RAP1_UNINDUCED_REP1_raw_fastqc.zip:md5,59452db3bb13c9b0553d08238999ef89",
            "RAP1_UNINDUCED_REP2_raw_fastqc.html:md5,d7f09428c8cb2af7657693f240909311",
            "RAP1_UNINDUCED_REP2_raw_fastqc.zip:md5,8b91abb96d701ccff806037d94e5df1d",
            "WT_REP1_raw_1_fastqc.html:md5,69a660d60b9d62c642a139b7d573d864",
            "WT_REP1_raw_1_fastqc.zip:md5,1f647b63c2127a95d2df865e9e2d9bc7",
            "WT_REP1_raw_2_fastqc.html:md5,51ee2fee5589e2e4697814d6d6f4204d",
            "WT_REP1_raw_2_fastqc.zip:md5,1bed4b9ee323facbcfc2612026535c5a",
            "WT_REP2_raw_1_fastqc.html:md5,aa156ab425f971a742cd607ece05e307",
            "WT_REP2_raw_1_fastqc.zip:md5,9eb10fd06e0e854ed9cedb5c18034931",
            "WT_REP2_raw_2_fastqc.html:md5,60ad56bc57db84ff236b4af22d7f9b7c",
            "WT_REP2_raw_2_fastqc.zip:md5,7725dfcabd958c3838e1b018bbd6948f"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-21T19:46:35.355529"
    }
}